app/models/cloud_database.rb

```diff
@@ -10,8 +10,41 @@ class CloudDatabase < ApplicationRecord
 
   serialize :extra_attributes
 
+  supports_not :create
   supports_not :delete
 
+  def self.create_cloud_database_queue(userid, ext_management_system, options = {})
+    task_opts = {
+      :action => "creating Cloud Database for user #{userid}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => 'CloudDatabase',
```

Comment:
Think you want to have this be `self.name` so it will be run on the subclass not the base class